src/main/java/com/github/mc1arke/sonarqube/plugin/scanner/ScannerConfigurationLoaderSensor.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 Artemy Osipov
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+package com.github.mc1arke.sonarqube.plugin.scanner;
+
+import com.github.mc1arke.sonarqube.plugin.CommunityBranchPlugin;
+import com.github.mc1arke.sonarqube.plugin.ce.pullrequest.bitbucket.server.BitbucketServerPullRequestDecorator;
+import com.github.mc1arke.sonarqube.plugin.ce.pullrequest.github.v4.GraphqlCheckRunProvider;
+import com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.GitlabServerPullRequestDecorator;
+import com.google.common.collect.Sets;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+
+import java.util.Optional;
+import java.util.Set;
+
+public class ScannerConfigurationLoaderSensor implements Sensor {
+
+    private final Set<String> overridableParameters;
+
+    public ScannerConfigurationLoaderSensor() {
+        this(Sets.newHashSet(
+                CommunityBranchPlugin.PULL_REQUEST_PROVIDER,
+                BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_PROJECT_KEY,
+                BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_USER_SLUG,
+                BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_REPOSITORY_SLUG,
+                GraphqlCheckRunProvider.PULL_REQUEST_GITHUB_REPOSITORY,
+                GitlabServerPullRequestDecorator.PULLREQUEST_GITLAB_REPOSITORY_SLUG
```

Comment:
@artemy-osipov Can we also add the credentials to the parameter list so that each parameter is customizable from within the CLI?

I think from a CI/CD perspective this would make sense

Reply:
There was a discussion in #72  about the ability to override all parameters. However, the @mc1arke raised concerns about the insecurity of overriding credentials and URLs.